R/test-that.R

```diff
@@ -190,6 +190,12 @@ test_code <- function(test, code, env, default_reporter, skip_on_empty = TRUE) {
   old <- options(rlang_trace_top_env = test_env)[[1]]
   on.exit(options(rlang_trace_top_env = old), add = TRUE)
 
+  if (!interactive()) {
```

Comment:
Rebinding something in the base namespace makes me a bit nervous. Maybe we should skip this on CRAN?

Ooooh, or should we set `_R_CHECK_BROWSER_NONINTERACTIVE_ `? https://github.com/wch/r-source/blob/22dc90fa26e0509cf7ec39202d6cc3416841467f/src/main/main.c#L1363

Reply:
That is a much better solution!

Reply:
What's the right place to set that? https://github.com/r-lib/testthat/blob/f31b02446cdc6254566e7602ab5dbfded68cc538/R/local.R#L67-L72?

Reply:
I _think_ `local_test_context()`?